.github/workflows/ci-pr.yml

```diff
@@ -59,29 +48,23 @@ jobs:
         uses: Swatinem/rust-cache@v1
       - name: Checkout
         uses: actions/checkout@v2
-      - name: Compile
-        run: cargo test --no-run ${{ env.FLAGS }}
-      - name: Test
-        run: cargo test ${{ env.FLAGS }}
-  msrv:
-    name: "Check MSRV: 1.54.0"
-    runs-on: ubuntu-latest
-    steps:
-      - name: Install rust
-        uses: actions-rs/toolchain@v1
+      - name: Compile (Minimal)
+        uses: actions-rs/cargo@v1
         with:
-          profile: minimal
-          toolchain: 1.54.0  # MSRV
-          override: true
-      - name: Cache Builds
-        uses: Swatinem/rust-cache@v1
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Default features
-        run: cargo check --all-targets
-      - name: All features + Debug
-        run: cargo check --all-targets --features "wrap_help yaml regex unstable-replace unstable-multicall unstable-grouped debug"
-      - name: No features
-        run: cargo check --all-targets --no-default-features --features "std cargo"
-      - name: UI Tests
-        run: cargo test --package clap_derive -- ui
+          command: test
+          args: --no-run ${{ needs.args.outputs.minimal }}
+      - name: Test (Minimal)
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: ${{ needs.args.outputs.minimal }}
+      - name: Compile (Full)
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --no-run ${{ needs.args.outputs.full }}
+      - name: Test (Full)
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: ${{ needs.args.outputs.full }}
```

Comment:
Testing all of these in one job is the approach I originally took in #2802 but then backed away due to performance.  How come you are moving back to that approach?

Reply:
I saw that you moved them back in #2901 and thought maybe you wanted them back again. I will make them separate jobs.

Reply:
I didn't move them back, I made them part of the matrix.

Reply:
I stand corrected.